File: CS/ExtendedDrillThroughReport/OrderDetailsReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using DevExpress.XtraPrinting.Preview;
using System.Windows.Forms;
using System.Data;

namespace ExtendedDrillThroughReport {
    public partial class OrderDetailsReport : DevExpress.XtraReports.UI.XtraReport, IDrillThroughReportHelper {

        #region IDrillThroughReportHelper
        public DocumentViewer Viewer { get; set; }
        public XtraReport ParentReport { get; set; }
        #endregion

        public OrderDetailsReport() {
            InitializeComponent();
        }

        public OrderDetailsReport(XtraReport parentReport, DocumentViewer viewer) {
            InitializeComponent();
            this.ParentReport = parentReport;
            this.Viewer = viewer;
        }

        private void xrTableCell1_PreviewMouseMove(object sender, PreviewMouseEventArgs e) {
            e.PreviewControl.Cursor = Cursors.Hand;
        }

        private void xrTableCell1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRLabel)sender).Tag = GetCurrentRow();
        }

        private void xrTableCell1_PreviewClick(object sender, PreviewMouseEventArgs e) {
            OrdersReport ordersReport = new OrdersReport(this, Viewer);
            ordersReport.Parameters["OrderID"].Value = (int)((DataRowView)e.Brick.Value).Row["OrderID"];
            ordersReport.CreateDocument();
            Viewer.DocumentSource = ordersReport;
        }

    }
}

File: VB/ExtendedDrillThroughReport/OrderDetailsReport.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraPrinting.Preview
Imports System.Windows.Forms
Imports System.Data

Namespace ExtendedDrillThroughReport
	Partial Public Class OrderDetailsReport
		Inherits DevExpress.XtraReports.UI.XtraReport
		Implements IDrillThroughReportHelper

		#Region "IDrillThroughReportHelper"
		Public Property Viewer() As DocumentViewer Implements IDrillThroughReportHelper.Viewer
		Public Property ParentReport() As XtraReport Implements IDrillThroughReportHelper.ParentReport
		#End Region

		Public Sub New()
			InitializeComponent()
		End Sub

		Public Sub New(ByVal parentReport As XtraReport, ByVal viewer As DocumentViewer)
			InitializeComponent()
			Me.ParentReport = parentReport
			Me.Viewer = viewer
		End Sub

		Private Sub xrTableCell1_PreviewMouseMove(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrTableCell1.PreviewMouseMove
			e.PreviewControl.Cursor = Cursors.Hand
		End Sub

		Private Sub xrTableCell1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrTableCell1.BeforePrint
			DirectCast(sender, XRLabel).Tag = GetCurrentRow()
		End Sub

		Private Sub xrTableCell1_PreviewClick(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrTableCell1.PreviewClick
			Dim ordersReport As New OrdersReport(Me, Viewer)
			ordersReport.Parameters("OrderID").Value = DirectCast(DirectCast(e.Brick.Value, DataRowView).Row("OrderID"), Integer)
			ordersReport.CreateDocument()
			Viewer.DocumentSource = ordersReport
		End Sub

	End Class
End Namespace
